pkg/operator/status.go

```diff
@@ -163,6 +163,27 @@ const (
 	asExpectedReason = "AsExpected"
 )
 
+func (optr *Operator) clearDegradedStatus(task string) error {
+	co, err := optr.fetchClusterOperator()
+	if err != nil {
+		return err
+	}
+	if co == nil {
+		return nil
+	}
+	if cov1helpers.IsStatusConditionFalse(co.Status.Conditions, configv1.OperatorDegraded) {
+		return nil
+	}
+	degradedStatusCondition := cov1helpers.FindStatusCondition(co.Status.Conditions, configv1.OperatorDegraded)
+	if degradedStatusCondition == nil {
+		return nil
+	}
+	if degradedStatusCondition.Reason != task+"Failed" {
+		return nil
```

Comment:
is there anyway a legit degraded message will not have Failed at the end?

Reply:
would it be better to just check that task is contained in Reason? (I dunno, I'm wondering?)